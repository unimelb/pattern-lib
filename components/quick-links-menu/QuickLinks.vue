<template>
  <div
    :class="{
      [`ql-menu--l${level}`]: true,
      'ql-menu--pad': pad,
      'ql-menu--condensed': condensed,
    }"
    class="grid ql-menu">
    <div class="cell cell--desk-2of3 ql-menu__content">
      <slot name="content" />
      <div
        v-if="secondaryLinks && secondaryLinks.length"
        :class="{ [`ql-menu__secondary--2col`]: secondaryCols === 2 }"
        class="ql-menu__secondary">
        <QuickLinksSecondaryItem
          v-for="item in secondaryLinks"
          :key="item.id"
          :href="item.href"
          :icon="item.icon"
          :title="item.title"
          class="cell" />
      </div>
    </div>
    <div class="cell cell--desk-1of3">
      <nav
        v-if="menuLinks.length"
        class="ql-menu__nav">
        <slot name="menu-header" />
        <QuickLinksMenuItem
          v-for="item in menuLinks"
          :key="item.id"
          :href="item.href"
          :title="item.title"
          :truncate="item.truncate" />
      </nav>
    </div>
  </div>
</template>

<script>
import QuickLinksMenuItem from './QuickLinksMenuItem.vue';
import QuickLinksSecondaryItem from './QuickLinksSecondaryItem.vue';

export default {
  components: { QuickLinksMenuItem, QuickLinksSecondaryItem },
  props: {
    level: {
      type: String,
      default: '1',
    },
    pad: {
      type: Boolean,
      default: false,
    },
    condensed: {
      type: Boolean,
      default: false,
    },
    menuLinks: {
      type: Array,
      default: () => [],
    },
    secondaryLinks: {
      type: Array,
      default: () => [],
    },
    secondaryCols: {
      type: Number,
      default: 2,
    },
  },
};
</script>
