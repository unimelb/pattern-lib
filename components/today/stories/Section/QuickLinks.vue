<template>
  <SectionImage
    :img="{ src: 'https://via.placeholder.com/800x800', alt: 'Placeholder image' }">
    <h2
      class="heading-md shim-mb2"
      style="max-width: 15rem;">
      Get started with Melbourne
    </h2>
    <nav aria-label="Get started with Melbourne">
      <QuickLinksAlt
        :items="links"
        class="shim-mb1"
        no-inset />
    </nav>
  </SectionImage>
</template>

<script>
import SectionImage from 'components/section/SectionImage.vue';
import QuickLinksAlt from 'components/quick-links-menu/QuickLinksAlt.vue';
import SectionImageMd from './SectionImage.md';

export default {
  components: { SectionImage, QuickLinksAlt },
  readme: { custom: SectionImageMd, html: true, source: true },
  data() {
    return {
      links: [
        { title: 'Start your application', href: '#' },
        { title: 'Explore scholarships', href: '#' },
        { title: 'Find accommodation', href: '#' },
        { title: 'Alternate pathways to Melbourne University', href: '#' },
      ],
    };
  },
};
</script>
