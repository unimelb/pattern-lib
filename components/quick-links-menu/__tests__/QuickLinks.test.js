import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import QuickLinks from '../QuickLinks.vue';

expect.extend(toHaveNoViolations);

describe('QuickLinks', () => {
  it('should match snapshot', () => {
    const result = shallow(QuickLinks).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(QuickLinks).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
