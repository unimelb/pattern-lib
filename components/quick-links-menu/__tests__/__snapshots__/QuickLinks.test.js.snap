// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLinks should match snapshot 1`] = `
<div
  class="grid ql-menu ql-menu--l1"
>
  <div
    class="cell cell--desk-2of3 ql-menu__content"
  >
     
    <!---->
  </div>
   
  <div
    class="cell cell--desk-1of3"
  >
    <!---->
  </div>
</div>
`;
